Feature#111

ershovd opened the pull request:
In order to workaround the bug changes made to ViewExtensions
- add ConfigureAwait(false)
- add boolean check

The issue appears to be a bug in Xamarin Forms https://bugzilla.xamarin.com/show_bug.cgi?id=41016  On a double click on translate InvalidOperation Exception with "An attempt was made to transition a task to a final state when it had already completed." error occured.
In order to proper fix the issue it may be required to implement the checking for tasks completion state. I manual tests the issue with these changes, it seems to be gone. If not, will focus on implementation of TaskCompletion state and checking
It took me 50m.

j0nscalet commented:
@ershovd Is this as simple as disabling the button on the first press while the animation runs?

From the bug report...
![image](https://user-images.githubusercontent.com/1328795/27253142-badc20a2-5323-11e7-93f4-5b432c2e0274.png)

j0nscalet commented:
@ershovd regardless I think your workaround is fine. 👍

j0nscalet commented on new/QualityKitchen.Mobile/Views/ViewExtensions.cs:
```diff
@@ -11,18 +11,23 @@ public static class ViewExtensions
 		private static readonly double FirstShakeOffset = 30;
 		private static readonly double SubsequentShakeOffset = 5;
 
+		private static bool IsAnimating = false; 
```
@ershovd At first glance I'm not thrilled that this is static.What happens if the Shake method is called twice?

ershovd replied:
@j0nscalet nothing bad should happens. Static variable initliazed once, on the App Domain start. so this is effectively a singleton

j0nscalet replied:
Right but if you did two subsequent calls to the `Shake` method, with the second call starting before the first one is finishing then we'd have a problem because effectively IsAnimated would be `true` when it should be false.

Something like:
```
view.Shake() <--- imagine this took 200ms
await view.Shake() <--- IsAnimating is true which will cause animation not to trigger
```

I don't think it's such a big deal at this point but let's be as thorough as we can. What are your thoughts on what I've described?

ershovd replied:
well, from what I've understand of static: there is only one instance of the `IsAnimating` per assembly (since the class is also static - it is initalized at the start of the assembly). and CLR guarantees that any thread accessing this value will access only this, single instance. If the situation you've mention have occured - then the check `if (IsAnimating) ` will be true and the method will be skipped. The variable will be set to false at the end execution.
If you are worries about that, we can use more serious stuff to ensure thread-safety like  `volatile`, `lock` or mutex. But unless current implementation is not good enough, this may be an overkill.

j0nscalet commented:
@ershovd I tested this and the crash is still occurring. Can you add support to disable the button when while the Shake animation is occurring? I think it's the better approach to take, anyway.

What do you think?

ershovd commented:
@j0nscalet I've implemented changes you've requested, both for temperatureReading dialog and for flag tasks dialog, please take a look.
Took me 40min, updated pull request accordingly
